### PR TITLE
Add capability for simulated numbers and email addresses

### PR DESCRIPTION
**Pull request by imdadahad**
This provides capability to stop emails/sms messages being sent to a specific set of simulated addresses. This provides a means to test these addresses on live without billing. Once a call is made to send an email or sms with the API a simulated response (message body, template id etc.) is returned.

### CODE REVIEW

File: tests/app/notifications/rest/test_send_notification.py [hunk 1]
```diff
@@ -674,14 +674,11 @@ def test_should_not_return_html_in_body(notify_api, notify_db, notify_db_session
 def test_should_not_send_email_if_team_api_key_and_not_a_service_user(notify_api, sample_email_template, mocker):
     with notify_api.test_request_context(), notify_api.test_client() as client:
         mocker.patch('app.celery.provider_tasks.send_email_to_provider.apply_async')
-
         data = {
             'to': "not-someone-we-trust@email-address.com",
             'template': str(sample_email_template.id),
         }
 
-        # import pdb
```
Comment (servingUpAces) [inline tests/app/notifications/rest/test_send_notification.py]: Please remove the commented lines

Comment (servingUpAces) [inline tests/app/notifications/rest/test_send_notification.py]: Ignore me - that was red not green

File: tests/app/notifications/rest/test_send_notification.py [hunk 2]
```diff
@@ -944,3 +941,51 @@ def test_should_delete_sms_notification_and_return_error_if_sqs_fails(notify_api
...
+
+    auth_header = create_authorization_header(service_id=sample_email_template.service_id)
+
+    response = client.post(
+        path='/notifications/email',
+        data=json.dumps(data),
+        headers=[('Content-Type', 'application/json'), auth_header])
+
+    assert response.status_code == 201
+    apply_async.assert_not_called()
```
Comment (servingUpAces) [inline tests/app/notifications/rest/test_send_notification.py]: I think we should make sure the notification is not persisted

File: tests/app/notifications/rest/test_send_notification.py [hunk 3]
```diff
@@ -944,3 +941,51 @@ def test_should_delete_sms_notification_and_return_error_if_sqs_fails(notify_api
...
+
+    auth_header = create_authorization_header(service_id=sample_template.service_id)
+
+    response = client.post(
+        path='/notifications/sms',
+        data=json.dumps(data),
+        headers=[('Content-Type', 'application/json'), auth_header])
+
+    assert response.status_code == 201
+    apply_async.assert_not_called()
```
Comment (servingUpAces) [inline tests/app/notifications/rest/test_send_notification.py]: I think we should make sure the notification is not persisted

Comment (imdadahad) [inline tests/app/notifications/rest/test_send_notification.py]: Thanks, added additional checks.

File: tests/app/notifications/rest/test_send_notification.py [hunk 4]
```diff
@@ -944,3 +941,65 @@ def test_should_delete_sms_notification_and_return_error_if_sqs_fails(notify_api
...
+    auth_header = create_authorization_header(service_id=sample_email_template.service_id)
+
+    response = client.post(
+        path='/notifications/email',
+        data=json.dumps(data),
+        headers=[('Content-Type', 'application/json'), auth_header])
+
+    assert response.status_code == 201
+    assert not notifications_dao.get_notification_by_id(fake_uuid)
+    apply_async.assert_not_called()
```
Comment (servingUpAces) [inline tests/app/notifications/rest/test_send_notification.py]: You could assert Notification.query.count() == 0, then avoid the first mocker for create_uuid, shave a couple of lines of code.

**Comment by servingUpAces** [comment]
👍

**Pull request merged**